File: app/src/main/java/com/coryswainston/smart/dictionary/util/DictionaryNetworkUtils.java
package com.coryswainston.smart.dictionary.util;

import java.io.IOException;
import java.net.MalformedURLException;
import java.net.URL;

import javax.net.ssl.HttpsURLConnection;

/**
 * A wrapper for testing
 */

public class DictionaryNetworkUtils {
    public HttpsURLConnection connectToUrl(String path) throws IOException {
        return (HttpsURLConnection) (new URL(path).openConnection());
    }
}

File: app/src/main/java/com/coryswainston/smart/dictionary/util/DictionaryNetworkUtils.kt
package com.coryswainston.smart.dictionary.util

import java.io.IOException
import java.net.URL
import javax.net.ssl.HttpsURLConnection

/**
 * A wrapper for testing
 */
class DictionaryNetworkUtils {
    @Throws(IOException::class)
    fun connectToUrl(path: String?): HttpsURLConnection {
        return URL(path).openConnection() as HttpsURLConnection
    }
}